### PR TITLE
add missing funcs for th client

### PR DESCRIPTION
**Pull request by kvarekamp**
this pr adds funcs to the client sdk that are needed for the th client. They are currently injected into the sdk at runtime:

```
from thematic_client_sdk.data import Data
Data.download_concepts = download_concepts
Data.upload_concepts = upload_concepts
...

```
Which is a bit messy. Some of these may be useful for others too

I do think there are some admin only restrictions on some of these funcs but not sure which ones. If we merge this PR, prob best to properly document that?

### CODE REVIEW

File: thematic_client_sdk/data.py [hunk 1]
```diff
@@ -112,6 +112,35 @@ def download_data(self, download_location, survey_id, result_id=None, output_for
 
         return True
 
+    def upload_data(self, survey_id, file_location, job_type):
```
Comment (nshgraph) [inline thematic_client_sdk/data.py]: This seems to be the same as the first function in the file? (see line 7)

File: thematic_client_sdk/data.py [hunk 2]
```diff
@@ -157,3 +204,110 @@ def download_db(self, download_location, survey_id, result_id=None):
...
+            raise Exception("Could not retrieve user log: " + str(response.text))
+
+        with open(download_location, "wb") as f_handle:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:  # filter out keep-alive new chunks
+                    f_handle.write(chunk)
+
+        return True
+
+    def download_upload(self, survey_id, download_location, upload_id, unconverted):
```
Comment (nshgraph) [inline thematic_client_sdk/data.py]: This is in upload.py named  get_input()

File: thematic_client_sdk/data.py [hunk 3]
```diff
@@ -157,3 +204,110 @@ def download_db(self, download_location, survey_id, result_id=None):
...
+        if response.status_code != 200:
+            raise Exception("Could not upload concepts: " + str(response.text))
+
+        resp = response.json()
+        if not resp or "data" not in resp or "upload_id" not in resp["data"]:
+            raise Exception(f"upload concepts got a response that did not have the expected format: {resp}")
+        return resp["data"]["upload_id"]
+
+
+    def get_upload_status(self, survey_id, upload_id):
```
Comment (nshgraph) [inline thematic_client_sdk/data.py]: All the functions with upload_id as a parameter should really go in upload.py

**Review by nshgraph — CHANGES_REQUESTED** [review]

File: thematic_client_sdk/data.py (hunk 1 above)
Comment (kvarekamp) [reply thematic_client_sdk/data.py]: yes - removed

File: thematic_client_sdk/data.py (hunk 2 above)
Comment (kvarekamp) [reply thematic_client_sdk/data.py]: yes, removed

File: thematic_client_sdk/data.py (hunk 3 above)
Comment (kvarekamp) [reply thematic_client_sdk/data.py]: moved (and renamed) all upload related funcs to upload_jobs.py

**Pull request closed without merging**